components/DetailsDashboard.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import styles from './DetailsCSS.module.css';
+import DetailsDashboardList from './DetailsDashboardList';
+//getStudentProgressInSuperblock
+
+import { getStudentProgressInSuperblock } from '../util/api_proccesor';
+
+export default function DetailsDashboard(props) {
+  const printSuperblockTitle = individualSuperblockJSON => {
+    let indexOfTitleInSuperblockTitlesArray =
+      props.superblocksDetailsJSONArray.indexOf(individualSuperblockJSON);
+    let superblockTitle =
+      props.superblockTitles[indexOfTitleInSuperblockTitlesArray];
+    return superblockTitle;
+  };
+
+  const superblockProgress = superblockDashedName => {
+    let studentProgress = props.studentData;
+
+    return getStudentProgressInSuperblock(
+      studentProgress,
+      superblockDashedName
+    );
+  };
+
+  return (
+    <>
+      {props.superblocksDetailsJSONArray.map((arrayOfBlockObjs, idx) => {
+        let index = props.superblocksDetailsJSONArray.indexOf(arrayOfBlockObjs);
+        let superblockDashedName =
+          props.superblocksDetailsJSONArray[index][0].superblock;
+        let progressInBlocks = superblockProgress(superblockDashedName);
+        let superblockTitle = printSuperblockTitle(arrayOfBlockObjs);
+        return (
+          <div key={idx} className={styles.board_container}>
+            <DetailsDashboardList
+              superblockTitle={superblockTitle}
+              blockData={arrayOfBlockObjs}
+              studentProgressInBlocks={progressInBlocks}
+            ></DetailsDashboardList>
+          </div>
+        );
+      })}
```

Comment:
Is there an edge case here where if we do not have a `superblocksDetailsJSONArray` is undefined, or of length 0? We can add conditional rendering to check if our `superblocksDetailsJSONArray` is undefined/has a length of 0, display a message to alert the user that there is no data to display

Reply:
Hi @GuillermoFloresV this is a good separate issue for someone wanting to contribute to the UI/Design side of the page. I prefer to create a separate issue for conditional rendering.